src/main/kotlin/org/jetbrains/exposed/sql/Constraints.kt

```diff
@@ -76,17 +76,17 @@ data class CheckConstraint(val tableName: String, val checkName: String, val che
     }
 
     override fun createStatement(): List<String> {
-        if (currentDialect is MysqlDialect) throw UnsupportedOperationException("Check constraints are not currently supported by MySQL")
+        if (currentDialect is MysqlDialect) exposedLogger.warn("CHECK constraints are not currently supported by MySQL")
```

Comment:
One more "idea": why not to return an empty list in a case when the database doesn't support constraints? With warn logging ofc. 
Here and in `dropStatements()`. And you can remove any checks in `modifyStatements()`.